compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -951,3 +952,35 @@ fn has_expected_num_generic_args<'tcx>(
         generics.count() == expected + if generics.has_self { 1 } else { 0 }
     })
 }
+
+/// Suggests calling the constructor of a tuple struct or enum variant
+///
+/// * `snippet` - The snippet of code that references the constructor
+/// * `span` - The span of the snippet
+/// * `params` - The number of parameters the constructor accepts
+/// * `err` - A mutable diagnostic builder to add the suggestion to
+fn suggest_call_constructor(
+    snippet: impl std::fmt::Display,
+    span: Span,
+    params: usize,
+    err: &mut DiagnosticBuilder<'_>,
+) {
+    // Note: tuple-structs don't have named fields, so just use placeholders
+    let mut args = String::new();
```

Comment:
Is this more idiomatic?
```suggestion
    let args = vec!["_"; params].join(", ");
```

Reply:
Thanks for the suggestions, Tom. I've gone ahead and applied it

Reply:
Just noting that you'll have to remove the subsequent `for` loop as well.

Reply:
Of course. I've pushed a fix